File: VB/WindowsFormsApplication154/RepositoryItemCustomTextEdit.vb
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports System.Drawing


Namespace WindowsFormsApplication154
    <UserRepositoryItem("RegisterCustomTextEdit")> _
    Public Class RepositoryItemCustomTextEdit
        Inherits RepositoryItemTextEdit

        Shared Sub New()
            RegisterCustomTextEdit()
        End Sub

        Public Const CustomEditName As String = "CustomTextEdit"

        Public Sub New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return CustomEditName
            End Get
        End Property

        Public Shared Sub RegisterCustomTextEdit()
            Dim img As Image = Nothing
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(CustomEditName, GetType(CustomTextEdit), GetType(RepositoryItemCustomTextEdit), GetType(CustomTextEditViewInfo), New TextEditPainter(), True, img))
        End Sub

        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            BeginUpdate()
            Try
                MyBase.Assign(item)
                Dim source As RepositoryItemCustomTextEdit = TryCast(item, RepositoryItemCustomTextEdit)
                If source Is Nothing Then
                    Return
                End If
                '
            Finally
                EndUpdate()
            End Try
        End Sub
    End Class

End Namespace

File: CS/WindowsFormsApplication154/RepositoryItemCustomTextEdit.cs
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using System.Drawing;


namespace WindowsFormsApplication154 {
    [UserRepositoryItem("RegisterCustomTextEdit")]
    public class RepositoryItemCustomTextEdit : RepositoryItemTextEdit {
        static RepositoryItemCustomTextEdit() {
            RegisterCustomTextEdit();
        }

        public const string CustomEditName = "CustomTextEdit";

        public RepositoryItemCustomTextEdit() {
        }

        public override string EditorTypeName
        {
            get
            {
                return CustomEditName;
            }
        }

        public static void RegisterCustomTextEdit() {
            Image img = null;
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(CustomEditName, typeof(CustomTextEdit), typeof(RepositoryItemCustomTextEdit), typeof(CustomTextEditViewInfo), new TextEditPainter(), true, img));
        }

        public override void Assign(RepositoryItem item) {
            BeginUpdate();
            try {
                base.Assign(item);
                RepositoryItemCustomTextEdit source = item as RepositoryItemCustomTextEdit;
                if (source == null) return;
                //
            } finally {
                EndUpdate();
            }
        }
    }

}
